plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenManager.java

```diff
@@ -51,15 +53,14 @@ public Optional<AuthenticationToken> translateAuthToken(org.opensearch.identity.
             final BasicAuthToken basicAuthToken = (BasicAuthToken) authenticationToken;
             return Optional.of(new UsernamePasswordToken(basicAuthToken.getUser(), basicAuthToken.getPassword()));
         }
-
         return Optional.empty();
     }
 
     @Override
-    public AuthToken issueToken(String audience) {
+    public AuthToken issueOnBehalfOfToken(Map<String, Object> claims) {
 
         String password = generatePassword();
```

Comment:
The code in the Shiro identity plugin is tough to follow. In the security plugin, would the implementation of this method issue a token on behalf of the currently authenticated user?

Reply:
The code in shiro is basically taking the audience and using it for testing. I added a comment.